presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/CheckUnsupportedPrestissimoTypes.java

```diff
@@ -121,71 +130,81 @@ public Void visitValues(ValuesNode node, Void context)
+            String errorMessage = hasType(input.getType());
+            checkState(errorMessage == null, errorMessage);
             return null;
         }
 
         @Override
         public Void visitCall(CallExpression call, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(call.getType()), errorMessage);
+            String errorMessage = hasType(call.getType());
+            checkState(errorMessage == null, errorMessage);
             return super.visitCall(call, context);
         }
 
         @Override
         public Void visitSpecialForm(SpecialFormExpression specialForm, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(specialForm.getType()), errorMessage);
+            String errorMessage = hasType(specialForm.getType());
+            checkState(errorMessage == null, errorMessage);
             return super.visitSpecialForm(specialForm, context);
         }
 
         @Override
         public Void visitIntermediateFormExpression(IntermediateFormExpression expression, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(expression.getType()), errorMessage);
+            String errorMessage = hasType(expression.getType());
+            checkState(errorMessage == null, errorMessage);
             return super.visitIntermediateFormExpression(expression, context);
         }
     }
 
-    private static boolean hasTimestampWithTimezoneType(Type type)
+    private static String hasType(Type type)
     {
-        if (type.equals(TIMESTAMP_WITH_TIME_ZONE)) {
-            return true;
+        if (type.equals(TIMESTAMP_WITH_TIME_ZONE) && config.isDisableTimeStampWithTimeZoneForNative()) {
+            return "Timestamp with Timezone type is not supported in Prestissimo";
+        }
+        if (type.equals(IPADDRESS) && config.isDisableIPAddressForNative()) {
+            return "IPAddress type is not supported in Prestissimo";
         }
         if (type instanceof ArrayType) {
-            return hasTimestampWithTimezoneType(((ArrayType) type).getElementType());
+            return hasType(((ArrayType) type).getElementType());
         }
         else if (type instanceof MapType) {
-            return hasTimestampWithTimezoneType(((MapType) type).getKeyType()) || hasTimestampWithTimezoneType(((MapType) type).getValueType());
+            String key = hasType(((MapType) type).getKeyType());
+            return key != null ? key : hasType(((MapType) type).getValueType());
         }
         else if (type instanceof RowType) {
-            return ((RowType) type).getTypeParameters().stream().anyMatch(CheckNoTimestampWithTimezoneType::hasTimestampWithTimezoneType);
+            return ((RowType) type).getTypeParameters().stream().map(CheckUnsupportedPrestissimoTypes::hasType).filter(Objects::nonNull).findFirst().orElse(null);
         }
-        return false;
+        return null;
```

Comment:
Consider use Optional instead of return null

Reply:
Done